CMakeLists.txt

```diff
@@ -76,6 +82,51 @@ add_compile_options(-Wall -Wextra -W -Wshadow -Wformat -Wno-nonnull-compare
 message(STATUS "Requested multimedia engine: ${USE_MULTIMEDIA}")
 message(STATUS "Requested build mode: ${CMAKE_BUILD_TYPE}")
 
+if(APPLE)
+  macro(_get_brew_prefix BREW_RECIPE_NAME VAR)
+    execute_process(
+      COMMAND "${BREW}" --prefix ${BREW_RECIPE_NAME}
+      OUTPUT_VARIABLE MAC_${VAR}_PREFIX
+      RESULT_VARIABLE MAC_${VAR}_RESULT
+    )
+
+    string(STRIP "${MAC_${VAR}_PREFIX}" MAC_${VAR}_PREFIX)
+  endmacro()
+
+  # ncurses version shipped with the latest macOS SDK is a fossilware version of 5.7 (A.D. 2008)
+  # We need 6.1, we need to take it from brew (for instance) and we need to be
+  # sneaky since the newer versions must not disrupt the fossilware installed with
+  # the SDK
+
+  # We support homebrew (https://brew.sh/) for now since I don't use anything else
+  find_program(BREW brew REQUIRED)
+
+  # We set up the environment so that pkg-config below can find ncurses
```

Comment:
this seems kinda heavyweight for what we need. is it not sufficient to add /usr/local/include and /usr/local/lib to the include/link paths in CMake, and then use the same routines we do for everything else? i really hate the idea of updating things in two places, especially in shitty shitty CMake

Reply:
i'll see what can be done, but if you have particular reasoning why you went this direction, i'd like to hear it

Reply:
macOS ships with old (sometimes ancient) versions of software such as ncurses, bash, etc.

In order not to conflict with the stuff that ships with the OS, `brew` organizes things in `/usr/local/opt` (which is a collection of symlinks to specific versions in `/usr/local/Cellar`), and you get these kind of messages with `brew info`:

```
$ brew info ncurses
...

ncurses is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have ncurses first in your PATH, run:
  echo 'export PATH="/usr/local/opt/ncurses/bin:$PATH"' >> /Users/michael/.bash_profile

For compilers to find ncurses you may need to set:
  export LDFLAGS="-L/usr/local/opt/ncurses/lib"
  export CPPFLAGS="-I/usr/local/opt/ncurses/include"

For pkg-config to find ncurses you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/ncurses/lib/pkgconfig"

...
```

While it's not the only possibility, pretty much everyone doing open source on macOS consumes Homebrew packages and/or publishes such packages.

Reply:
Sorry if that's a repeat from something I explained in my emails; main point is that on macOS it's more likely to work more easily for more devs and users if its done the Homebrew way.